Classes/Wwwision/Neos/MailChimp/Domain/Service/MailChimpService.php

```diff
@@ -36,27 +37,39 @@ class MailChimpService
      */
     private $apiEndpoint;
 
+    /**
+     * @Flow\Inject
+     * @var UriFactoryInterface
+     */
+    protected $uriFactory;
+
+    /**
+     * @Flow\Inject
+     * @var ServerRequestFactoryInterface
```

Comment:
Should that be `RequestFactoryInterface` (i.e. no `Server…`?)

Reply:
But why? `RequestEngineInterface::sendRequest()` expects an instance of `ServerRequestInterface` (see https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Classes/Http/Client/RequestEngineInterface.php#L31)

Reply:
Doh. Why, though? Maybe the author, @albe, knows? Because if I remember correctly, it was him that pointed me into the direction that made me comment on this: A "server request" is what comes in, not what you send out.

Reply:
I thought so as well

Reply:
That's indeed strange, but: the request engine wasn't built with the distinction of in/out request in mind. It just takes the request it gets and pretends that was the request that came in from somewhere. Optimally, it should take a Request (outgoing), then the internal request engine converts that into a ServerRequest and gives that to the application. Missed opportunity to API refactoring :/

Reply:
The RequestEngine doesn't convert anything. The InternalRequestEngine, maybe, yes. But the CurlEngine actually sends a requesst somewhere and returns the response. Anyway, isn't that something we should fix? Not here, of course, but generally speaking…

Reply:
> Anyway, isn't that something we should fix?

Yes.. but with which version? Flow 7.0? :-/

Reply:
I'd even dare to do it for 6.x – because that "was broken" only with 6.0.0. Before that there was no `ServerRequestInterface`, right?

Reply:
Okay, after a quick glance, it's not as easy. The main issue is following:

`Browser::request()` expects to be able to set things like uploaded files and the `$server` variables. This is not doable with a non-ServerRequest. We could somehow work around that FileUpload thing, by serializing it into a HTTP body with proper boundaries and all that (ugh). But even then, the sender would not be able to set server params, if `sendRequest` doesn't either take a `ServerRequest` as it currently does, or is extended with an additional parameter `$server` that is only really used by the internal request engine.

As far as I can tell, at least in Flow, that `$server` parameter isn't really used anywhere though. So we could strip that from `request()`, but that would be a breaking change.

Anyway, seeing how this is a little bit bigger change than anticipated, I'd go for fixing this as is and use `ServerRequest` for now.

Reply:
I gave it a try: https://github.com/neos/flow-development-collection/pull/1791